### PR TITLE
chore: cleaning up our tox, makefile and CI

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR increases our usage of tox and cleans up the Makefile, in an effort to improve our DevX.

## Fixes

-

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

-

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ SERVICE_ID := "${SERVICE_ID}"
 PLATFORM_STR := $(shell uname)
 
 .PHONY: clean
-clean: clean-build clean-pyc clean-test clean-docs
+clean: clean-test clean-build clean-pyc clean-docs
```
Comment (DavidMinarsch) [inline Makefile:7]: First let's remove large folders, just reordering

File: Makefile [hunk 2]
```diff
@@ -71,19 +71,12 @@ security:
 # update copyright headers
 .PHONY: generators
 generators:
+	tox -e abci-docstrings
```
Comment (DavidMinarsch) [inline Makefile:74]: Let's add another generator

File: Makefile [hunk 3]
```diff
@@ -100,41 +93,6 @@ common-checks-2:
 copyright:
 	python scripts/check_copyright.py
 
-.PHONY: check-copyright
```
Comment (DavidMinarsch) [inline Makefile:103]: All the below are duplications and also don't use tox. Going forward let's just use tox to have max consistency.

File: Makefile [hunk 4]
```diff
@@ -148,14 +106,7 @@ test:
...
 .PHONY: all-checks
-all-checks:
-	make clean \
-	&& make formatters \
-	&& make code-checks \
-	&& make security \
-	&& make generators \
-	&& make common-checks-1 \
-	&& make common-checks-2
+all-checks: clean formatters code-checks security make generators make common-checks-1 make common-checks-2
```
Comment (DavidMinarsch) [inline Makefile:109]: simpler

File: Makefile [hunk 5]
```diff
@@ -411,8 +362,8 @@ teardown-kubernetes:
 	kubectl delete ns ${VERSION}
 	echo "Done!"
 
-.PHONY: check_abci_specs
-check_abci_specs:
+.PHONY: generate_abci_specs
```
Comment (DavidMinarsch) [inline Makefile:365]: correct name now

File: tox.ini [hunk 6]
```diff
@@ -6,8 +6,9 @@
 ; we set the associated flag (e.g. for linting we don't need
 ; the package installation).
 [tox]
-envlist = bandit, safety, black, black-check, isort, isort-check, check-copyright, docs, flake8, liccheck, mypy, py{3.6,3.7,3.8,3.9}
-
+envlist = bandit, safety, black, black-check, isort, isort-check, check-copyright, check-hash, check-packages, check-pipfiles, check-api-docs, docs, check-doc-links-hashes, flake8, mypy, pylint, darglint, vulture, check-generate-all-protocols, check-abciapp-specs, check-abci-docstrings, check-handlers, py{3.7,3.8,3.9,3.10}-{win,linux,darwin}
+; when running locally we don't want to fail for no good reason
+skip_missing_interpreters = true
```
Comment (DavidMinarsch) [inline tox.ini:11]: very useful!

File: tox.ini [hunk 7]
```diff
@@ -60,10 +61,10 @@ setenv =
 commands =
     pytest -rfE --doctest-modules autonomy tests/ --cov=autonomy --cov-report=html --cov=packages/valory --cov-report=xml --cov-report=term --cov-report=term-missing --cov-config=.coveragerc {posargs}
 
-
-[testenv:py3.10]
+[testenv:py3.10-linux]
 basepython = python3.10
 usedevelop = True
+platform=^linux$
```
Comment (DavidMinarsch) [inline tox.ini:67]: this gets evaluated against sys.platform and so `tox` works out of the box with the platform specific commands

File: tox.ini [hunk 8]
```diff
@@ -303,7 +312,7 @@ deps =
     flake8-eradicate==1.1.0
     flake8-isort==4.0.0
     pydocstyle==6.1.1
-commands = flake8 autonomy packages/valory scripts tests
+commands = flake8 autonomy packages/valory scripts tests deployments
```
Comment (DavidMinarsch) [inline tox.ini:315]: one missing

File: tox.ini [hunk 9]
```diff
@@ -313,7 +322,7 @@ deps =
     mypy==0.910
 commands = mypy autonomy packages/valory --disallow-untyped-defs
            mypy tests --disallow-untyped-defs
-           mypy scripts
+           mypy scripts --disallow-untyped-defs
```
Comment (DavidMinarsch) [inline tox.ini:325]: missing relative to make command

File: tox.ini [hunk 10]
```diff
@@ -331,8 +340,7 @@ deps =
     open-aea==1.16.0
     open-aea-ledger-ethereum===1.16.0.post1
     open-aea-ledger-cosmos===1.16.0
-
-commands = pylint autonomy packages/valory scripts
+commands = pylint autonomy packages/valory scripts deployments
```
Comment (DavidMinarsch) [inline tox.ini:343]: one missing

File: tox.ini [hunk 11]
```diff
@@ -353,7 +361,7 @@ skipsdist = True
 skip_install = True
 deps =
     darglint==1.8.0
-commands = 	darglint autonomy scripts packages/valory/* tests
+commands = 	darglint autonomy scripts packages/valory/* tests deployments
```
Comment (DavidMinarsch) [inline tox.ini:364]: one missing

File: tox.ini [hunk 12]
```diff
@@ -371,6 +379,21 @@ usedevelop = True
 deps =
 commands = {toxinidir}/scripts/spell-check.sh
 
+[testenv:abci-docstrings]
```
Comment (DavidMinarsch) [inline tox.ini:382]: generator

File: Pipfile [hunk 13]
```diff
@@ -47,6 +47,7 @@ pytest = "==7.0.0"
 pytest-asyncio = "==0.18.0"
 pytest-cov = "==3.0.0"
 pytest-randomly = "==3.11.0"
+pytest-rerunfailures = "==10.0"
```
Comment (DavidMinarsch) [inline Pipfile:50]: this was missing but flaky test markers were used :/

File: docs/abci.md [hunk 14]
```diff
@@ -2,11 +2,11 @@
 
 
 
-The [Application BlockChain Interface (ABCI)](https://docs.tendermint.com/master/spec/abci/) defines the boundary between the consensus engine (the blockchain) and an application to be replicated across a number of platforms.
```
Comment (DavidMinarsch) [inline docs/abci.md:5]: Tendermint changed urls, they are no longer available in some cases

**Review by dvilelaf — DISMISSED** [review]
LGTM, waiting for tests.

File: docs/abci.md (hunk 14 above)
Comment (angrybayblade) [reply docs/abci.md:5]: Why is URL checker not failing for this URL?

Comment (DavidMinarsch) [reply docs/abci.md:5]: It did locally - it was disabled in CI (enabling now!)

File: .github/workflows/main_workflow.yml [hunk 15]
```diff
@@ -284,26 +284,26 @@ jobs:
...
-          cd third_party/safe-contracts &&\
-            sudo chown -R $(whoami) .husky &&\
-            yarn
-          cd ../..
-
-          # build hardhat node in third_party/contracts-amm
-          cd third_party/contracts-amm &&\
-            yarn
-          cd ../..
+          # cd third_party/safe-contracts &&\
```
Comment (DavidMinarsch) [inline .github/workflows/main_workflow.yml:287]: Keeping it here for ease of reference, but we don't need it for mac and windows!

File: Makefile (hunk 4 above)
Comment (angrybayblade) [inline Makefile:109]: all-checks: clean formatters code-checks security generators common-checks-1 common-checks-2

**Review by angrybayblade — CHANGES_REQUESTED** [review]

File: Makefile (hunk 4 above)
Comment (angrybayblade) [inline Makefile:109]: Remove the extra `make` statements

Comment (DavidMinarsch) [reply Makefile]: fixed in [d8e7885](https://github.com/valory-xyz/open-autonomy/pull/1214/commits/d8e7885f22025637be39086a427ded915e4b76ad)

File: Makefile [hunk 16]
```diff
@@ -71,19 +72,12 @@ security:
 # update copyright headers
 .PHONY: generators
 generators:
+	tox -e abci-docstrings
+	python scripts/check_copyright.py
```
Comment (DavidMinarsch) [inline Makefile:76]: this affects hashing potentially, so needs to come first

File: tests/test_agents/test_register_reset_abci.py [hunk 17]
```diff
@@ -37,7 +37,7 @@
 
 @pytest.mark.e2e
 @pytest.mark.integration
-@pytest.mark.flaky(reruns=3)
+@pytest.mark.flaky(reruns=1)
```
Comment (dvilelaf) [inline tests/test_agents/test_register_reset_abci.py:40]: Why this change?

**Review by angrybayblade — APPROVED** [review]

File: tests/test_agents/test_register_reset_abci.py (hunk 17 above)
Comment (Karrenbelt) [inline tests/test_agents/test_register_reset_abci.py:40]: Why reduced?

Comment (dvilelaf) [reply tests/test_agents/test_register_reset_abci.py:40]: https://github.com/valory-xyz/open-autonomy/pull/1214#discussion_r951578733

**Review by dvilelaf — APPROVED** [review]
